site/app/libraries/database/DatabaseQueries.php

```diff
@@ -7471,13 +7471,22 @@ public function getGradeableAccessTeam(
 
     public function getUserGroups(string $user_id): array {
         $this->submitty_db->query(
-            'SELECT DISTINCT c.group_name FROM courses c INNER JOIN courses_users cu on c.course = cu.course AND 
+            'SELECT DISTINCT c.group_name FROM courses c INNER JOIN courses_users cu on c.course = cu.course AND
                    c.semester = cu.semester WHERE cu.user_id = ? AND user_group = 1',
             [$user_id]
         );
         return $this->submitty_db->rows();
     }
 
+    public function getOtherCoursesWithSameGroup(string $semester, string $course): array {
```

Comment:
Let's exclude the default group from this query.  That is, if I'm in a group with group='root' (I think that's our default?), don't let me run plagiarism against any other course, period.  If lots of courses are archived with group=root no one should be able to compare against them.